aptos-move/move-examples/rewards_pool/sources/rewards_pool.move

```diff
@@ -0,0 +1,288 @@
+        let reward_tokens = reward_tokens(rewards_pool);
+        let rewards = vector[];
+        let claimer_addr = signer::address_of(claimer);
+        let rewards_data = unchecked_mut_rewards_pool_data(&rewards_pool);
+        vector::for_each(reward_tokens, |reward_token| {
+            let reward = rewards(claimer_addr, rewards_data, reward_token, epoch);
+            let reward_store = simple_map::borrow(&rewards_data.reward_stores, &reward_token);
+            if (reward == 0) {
+                vector::push_back(
+                    &mut rewards,
+                    fungible_asset::zero(fungible_asset::store_metadata(reward_store.store)),
+                );
+            } else {
+                // Withdraw the reward from the corresponding store.
+                let store_signer = &object::generate_signer_for_extending(&reward_store.store_extend_ref);
+                vector::push_back(&mut rewards, fungible_asset::withdraw(store_signer, reward_store.store, reward));
+
+                // Update the remaining amount of rewards for the epoch.
+                let epoch_rewards = smart_table::borrow_mut(&mut rewards_data.epoch_rewards, epoch);
+                let total_token_rewards = simple_map::borrow_mut(&mut epoch_rewards.total_amounts, &reward_token);
+                *total_token_rewards = *total_token_rewards - reward;
+            };
+        });
+
+        // Remove the claimer's allocation in the epoch as they have now claimed all rewards for that epoch.
+        let epoch_rewards = smart_table::borrow_mut(&mut rewards_data.epoch_rewards, epoch);
+        let all_shares = pool_u64::shares(&epoch_rewards.claimer_pool, claimer_addr);
+        pool_u64::redeem_shares(&mut epoch_rewards.claimer_pool, claimer_addr, all_shares);
+
+        rewards
+    }
+
+    /// Add rewards to the specified rewards pool. This can be called with multiple reward tokens.
+    public fun add_rewards(
+        rewards_pool: Object<RewardsPool>,
+        fungible_assets: vector<FungibleAsset>,
+        epoch: u64,
+    ) acquires RewardsPool {
+        let rewards_data = unchecked_mut_rewards_pool_data(&rewards_pool);
+        let reward_stores = &rewards_data.reward_stores;
+        vector::for_each(fungible_assets, |fa| {
+            let amount = fungible_asset::amount(&fa);
+            let reward_token = fungible_asset::metadata_from_asset(&fa);
+            assert!(simple_map::contains_key(reward_stores, &reward_token), EREWARD_TOKEN_NOT_SUPPORTED);
+
+            // Deposit the rewards into the corresponding store.
+            let reward_store = simple_map::borrow(reward_stores, &reward_token);
+            fungible_asset::deposit(reward_store.store, fa);
+
+            // Update total amount of rewards for this token for this epoch.
+            let total_amounts = &mut epoch_rewards_or_default(&mut rewards_data.epoch_rewards, epoch).total_amounts;
+            if (simple_map::contains_key(total_amounts, &reward_token)) {
```

Comment:
missing borrow_mut_with_default in simple_map

Reply:
Yep. It's slightly tricky because the syntax in simple_map is very strange to avoid requiring droppable values (relatively to smart_table). May be @xindingw can look into this as he's working on improving simple_map separately anyway :)